src/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2.java

```diff
@@ -133,7 +134,13 @@ public void scan(HttpMessage msg, String param, String value) {
 			boolean attackWorked = false;
             HttpMessage msg2 = getNewMsg();
 			setParameter(msg2, param, Constant.getEyeCatcher());
-            sendAndReceive(msg2);
+            try {
+    			sendAndReceive(msg2);
+        	} catch (InvalidRedirectLocationException|UnknownHostException e) {
+        		// Not an error, just means we probably attacked the redirect location

```

Comment:
So I just added on to the existing handling in this one:
https://github.com/kingthorin/zap-extensions/blob/328da95f752534260711afc3e42d5857e0361007/src/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2.java#L110-L111
trusting the original author's handling.

I guess at 165/166 in the new code we could break out if things aren't working.